### ISSUE TITLE
[`prefer-node-protocol`]: does not flag `typeof import("...")`

### ISSUE THREAD

**Issue by sadan4** [sindresorhus/eslint-plugin-unicorn#2773]
prefer-node-protocol does not check `typeof import("...")`

```ts
const assert = require("node:assert") as typeof import("assert");
//                                                      ^^^^^^
// This should be flagged as requiring the node protocol
```

I would be willing to file a PR to fix this.
